pkg/controller/sync/resource.go

```diff
@@ -165,6 +165,15 @@ func (r *federatedResource) ObjectForCluster(clusterName string) (*unstructured.
 	}
 	obj := &unstructured.Unstructured{Object: templateBody}
 
+	if len(obj.GetAnnotations()) > 0 {
+		r.RecordError("AnnotationsNotSupported", errors.New("metadata.annotations will not be set to avoid conflicting with local controllers."))
```

Comment:
This message seems misunderstanding. Actually, `metadata.annotations` still can be set. But by `override` instead of by `template`

Reply:
Updated. PTAL